File: #0647.palindromic-substrings.cpp
class Solution {
public:
    int countSubstrings(string s) {

        int size = s.size();
        int result = 0;
        for (int i = 0; i < 2 * size -1; i++) {
            int left = i / 2;
            int right = i / 2 + i % 2;
            while (left >= 0 && right < size && s[left] == s[right]) {
                result++;
                left--;
                right++;
            }
        }
        return result;
    }
};

File: #0647.palindromic-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countSubstrings = function(s) {
    let result = 0;
    for (let i = 0; i < s.length; i++) {
        for (let j = i, k = i; s[j] && s[j]==s[k]; j--,k++) {
            result++;
        }
        for (let j = i, k = i + 1; s[j] && s[j] == s[k]; j--, k++) {
            result++;
        }
    }
    return result;
};
